sdk/src/state_machine/zk/traits.rs

```diff
@@ -13,14 +12,17 @@ pub trait ZkvmHost: Zkvm {
 /// A Zk proof system capable of proving and verifying arbitrary Rust code
 /// Must support recursive proofs.
 pub trait Zkvm {
-    type CodeCommitment: Matches<Self::CodeCommitment> + Clone;
-    type Proof: ProofTrait<Self>;
+    type CodeCommitment: Matches<Self::CodeCommitment>
+        + Clone
+        + Debug
+        + Serialize
+        + DeserializeOwned;
     type Error: Debug;
 
-    fn verify(
-        proof: Self::Proof,
+    fn verify<'a>(
+        serialized_proof: &'a [u8],
         code_commitment: &Self::CodeCommitment,
-    ) -> Result<<<Self as Zkvm>::Proof as ProofTrait<Self>>::Output, Self::Error>;
+    ) -> Result<&'a [u8], Self::Error>;
```

Comment:
Since we do not use ProofTrait anymore, could you please add description, what this method returns in Ok()?

Reply:
Added